instrumentation/buildtime/mapping-file/plugin/src/main/java/com/splunk/rum/mappingfile/plugin/utils/MappingFileUploader.kt

```diff
@@ -0,0 +1,156 @@
+ */
+
+package com.splunk.rum.mappingfile.plugin.utils
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.splunk.rum.mappingfile.plugin.SplunkRumExtension
+import java.io.File
+import org.gradle.api.Project
+
+class MappingFileUploader(private val project: Project) {
+
+    private val httpClient = MappingFileUploadClient(project.logger)
+
+    fun uploadAfterBuild(variant: ApplicationVariant, buildId: String, extension: SplunkRumExtension) {
+        project.logger.info("Splunk RUM: Starting upload process for variant '${variant.name}'")
+
+        val accessToken = resolveAccessToken(extension)
+        val realm = resolveRealm(extension)
+
+        val missingConfigs = buildList {
+            if (accessToken.isBlank()) add("accessToken")
+            if (realm.isBlank()) add("realm")
+        }
+
+        if (missingConfigs.isNotEmpty()) {
+            project.logger.warn("Splunk RUM: Missing required configuration: ${missingConfigs.joinToString(", ")}")
+            project.logger.info("Splunk RUM: Configure in your build.gradle:")
+            project.logger.info(
+                """
+        splunkRum {
+            ${if ("accessToken" in missingConfigs) "accessToken = \"your-token-here\"" else ""}
+            ${if ("realm" in missingConfigs) "realm = \"your-realm-here\"" else ""}
+        }
+                """.trimIndent()
+            )
+            project.logger.info("Splunk RUM: Or use environment variables: SPLUNK_ACCESS_TOKEN, SPLUNK_REALM")
+            return
+        }
+
+        project.logger.debug(
+            "Splunk RUM: Configuration resolved - realm: '$realm', token configured: ${accessToken.isNotBlank()}"
+        )
+
+        val mappingFile = findMappingFile(variant)
+        if (mappingFile == null) {
+            project.logger.error(
+                """
+                Splunk RUM: Mapping file not found for variant '${variant.name}'
+                
+                Searched locations:
+                ${getMappingFileLocations(variant).joinToString("\n") { "  - ${it.absolutePath}" }}
+                
+                This usually means:
+                1. ProGuard/R8 minification is not properly configured
+                2. The build did not complete successfully
+                3. The mapping file is in a custom location
+                                
+                Please verify that minification is enabled and the build completed without errors.
+                If the mapping file is in a custom location, please disable the uploadEnabled and upload it manually.
+                """.trimIndent()
+            )
+            return
+        }
+
+        project.logger.info(
+            "Splunk RUM: Found mapping file (${mappingFile.length()} bytes) at: ${mappingFile.absolutePath}"
+        )
+
+        project.logger.lifecycle("Splunk RUM: Uploading mapping file for variant ${variant.name}")
+        project.logger.info("Splunk RUM: Upload details:")
```

Comment:
Nit: Replace with a single log and a multiline string, as you did previously for the logs above.